pkg/vault/vault.go

```diff
@@ -17,6 +17,44 @@ func NewVaultToken(vc *vaultClient) (string, error) {
 	return NewAuthClient(vc.config).GetVaultToken(vc)
 }
 
+// ListEngines fills a map with the secrets engines pulled from Vault.
+func ListEngines(ctx context.Context, path string) ([]string, error) {
+	environment := os.Getenv("ENVIRONMENT")
+
+	if environment == "production" {
+		log.SetFormatter(&log.JSONFormatter{})
+		log.SetLevel(log.WarnLevel)
+	} else {
+		log.SetFormatter(&log.TextFormatter{})
+		log.SetLevel(log.TraceLevel)
+	}
+
+	config, err := loadVaultEnvironment()
+	if err != nil {
+		return nil, fmt.Errorf("load client environment: %v", err)
+	}
```

Comment:
you can DRY this up with this helper method I added
```suggestion
	config, err := getConfig()
	if err != nil {
		return err
	}
```

Reply:
Thank you!!